### PR TITLE
[FIX] account: journal_id value change in account payment register wizzard

### PR DESCRIPTION
**Pull request by jbw-odoo**
journal_id value would change when grouping payments in wizard if
partner_bank_id was set on a move

### CODE REVIEW

**Review by william-andre — COMMENTED** [review]
The commit message is not clear at all.

**Comment by jbw-odoo** [comment]
> The commit message is not clear at all.

changed

**Comment by jbw-odoo** [comment]
@william-andre moved code to _compute_journal_id

**Comment by jbw-odoo** [comment]
@william-andre moved logic to journal_id compute method

File: addons/account/wizard/account_payment_register.py [hunk 1]
```diff
@@ -266,13 +243,34 @@ def _compute_group_payment(self):
...
+                lines = wizard.line_ids._origin
+                partner_bank_id = lines.move_id.mapped('partner_bank_id')
+
+                company_domain = [('company_id', '=', wizard.company_id.id)]
+                bank_domain = [('bank_account_id', '=', partner_bank_id.id), ('type', '=', 'bank')] if len(partner_bank_id) == 1 else None
+                no_bank_domain = [('type', 'in', ('bank', 'cash'))]
+                currency_domain = [('currency_id', '=', wizard.source_currency_id.id)] if wizard.source_currency_id else None
+
+                journal = None
+                if currency_domain:
```
Comment (william-andre) [inline addons/account/wizard/account_payment_register.py:261]:
```suggestion

                journal = None
                if wizard.source_currency_id:
                    currency_domain = [('currency_id', '=', wizard.source_currency_id.id)]

File: addons/account/wizard/account_payment_register.py [hunk 2]
```diff
@@ -266,13 +243,34 @@ def _compute_group_payment(self):
...
 
-    @api.depends('company_id', 'source_currency_id')
+    @api.depends('company_id', 'source_currency_id', 'line_ids')
     def _compute_journal_id(self):
         for wizard in self:
             if wizard.journal_id:
                 wizard.journal_id = wizard.journal_id
             else:
-                wizard.journal_id = self._search_journal(wizard.company_id.id, wizard.source_currency_id.id)
+                lines = wizard.line_ids._origin
```
Comment (william-andre) [inline addons/account/wizard/account_payment_register.py:252]: Why?

**Comment by jbw-odoo** [comment]
Pushed changes ...

File: addons/account/wizard/account_payment_register.py [hunk 3]
```diff
@@ -266,13 +243,33 @@ def _compute_group_payment(self):
...
+                    if bank_domain:
+                        journal = self.env['account.journal'].search(company_domain + currency_domain + bank_domain, limit=1)
+                    if not journal:
+                        journal = self.env['account.journal'].search(company_domain + currency_domain + no_bank_domain, limit=1)
+                if not journal and bank_domain:
+                    journal = self.env['account.journal'].search(company_domain + bank_domain, limit=1)
+                if not journal:
+                    journal = self.env['account.journal'].search(company_domain + no_bank_domain, limit=1)
+                if not journal:
+                    journal = self.env['account.journal'].search(company_domain, limit=1)
```
Comment (william-andre) [inline addons/account/wizard/account_payment_register.py:270]: You never want to have a journal that is neither bank neither cash

**Review by william-andre — COMMENTED** [review]
The logic seemed easier earlier

File: addons/account/wizard/account_payment_register.py (hunk 3 above)
Comment (jbw-odoo) [reply addons/account/wizard/account_payment_register.py]: removed

**Comment by jbw-odoo** [comment]
> The logic seemed easier earlier

It was, but it would not have suggested a journal_id if a partner_bank_id was set on the move, but that partner_bank_id was not the bank_account_id of a journal. Not the most usual case, but still.

**Comment by jbw-odoo** [comment]
@william-andre r+ ? 😶

File: addons/account/wizard/account_payment_register.py [hunk 4]
```diff
@@ -266,13 +243,31 @@ def _compute_group_payment(self):
...
+
+                journal = None
+                if wizard.source_currency_id:
+                    currency_domain = [('currency_id', '=', wizard.source_currency_id.id)]
+                    if bank_domain:
+                        journal = self.env['account.journal'].search(company_domain + currency_domain + bank_domain, limit=1)
+                    if not journal:
+                        journal = self.env['account.journal'].search(company_domain + currency_domain + no_bank_domain, limit=1)
+                if not journal and bank_domain:
+                    journal = self.env['account.journal'].search(company_domain + bank_domain, limit=1)
```
Comment (william-andre) [inline addons/account/wizard/account_payment_register.py:266]: I'm not sure which should have the highest priority; the currency or the bank.

Both make sense kind of sense so lets go with this.

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**